### ISSUE TITLE
Minify: JQuery isn't moved to footer

### ISSUE THREAD

**Issue by nigrosimone** [szepeviktor/fix-w3tc#391]
in this page http://sndsgn.altervista.org/about-us/

jquery is minified alone in the file http://sndsgn.altervista.org/wp-content/cache/minify/c7035.js?x81922

but isn't moved to footer for some unknown reason and cause render blocking error

![immagine](https://cloud.githubusercontent.com/assets/310077/23043485/b8ffdb1a-f49c-11e6-8db8-7b2441e3908d.jpg)

my minify settings are:

![immagine](https://cloud.githubusercontent.com/assets/310077/23043641/5101354e-f49d-11e6-8da1-d8859ff206e1.jpg)

theme Avada (premium)

What can cause this issue?

**Comment by amiga-500** [comment]
Its been a while so i will have to investigate this later (probably in a day or two) but if i recall jQuery is very finicky about being moved to the footer.  I believe it's due to the _jquery-mingrate.js_ file that wp forceably adds.  For me, I had to forceably remove jquery via `wp_dequeue_script()` from the wp stack and readd it into the stack before w3tc was able to successfully move it to the footer.  Again, its been so long so i'd need a refresher.  Still swamped with other bugs so can't check for a while.

In the meantime here are a list of W3TC's macro comment tags.  When they are placed in your html they will put the js/css minified files there:

```php
<!-- W3TC-include-js-head -->
<!-- W3TC-include-js-body-start -->
<!-- W3TC-include-js-body-end -->
<!-- W3TC-include-css -->
```
In your case, i figure **`<!-- W3TC-include-js-head -->`** is what you need and placed in your theme's _footer.php_ file

When using Manual mode for Minify this isn't a problem.  Just select jquery to "_Embed before </body>_" and it works.

**Comment by nigrosimone** [comment]
i try it, i moved `<!-- W3TC-include-js-head -->` in `footer.php` but the issue persists.

**Comment by amiga-500** [comment]
Ok thanks for the info.

**Comment by amiga-500** [comment]
You can resolve your issue via adding the following into your theme's **_functions.php_** file:

```php
add_action('wp_print_scripts','my_dequeue_scripts',99);
add_action('wp_enqueue_scripts','my_enqueue_scripts',99);

function my_dequeue_scripts()
{
	wp_deregister_script("jquery");
	wp_enqueue_script("jquery-migrate",array("jquery-footer"),false,null,true);
}
function my_enqueue_scripts()
{
	wp_enqueue_script("jquery-footer",includes_url("/js/jquery/jquery.js"),false,null,true);
}
```

But i'm pretty sure this has always been the way it is...in that one either has to make use of the async/defer attributes or dequeuing/enqueuing to get around the render blocking issue caused by jQuery when using **Automatic** mode.

Oh and just an update...those macros i gave in my first reply are specific to **Manual** mode only.  The are ignored when using **Automatic** that's why it didn't work for you.

When using Automatic, W3TC simply minifies all files to the first js file location it finds which is usually in the header.  In Manual mode you have the option to decide where to put it.  Your "Operations in Areas" selection only affects javasript files that happen to be in those areas.  In your case, since jQuery is in the header it is set to be in blocking mode -- it wont be "defer" since you have it set to "Default (blocking").

**Comment by nigrosimone** [comment]
> When using Automatic, W3TC simply minifies all files to the first js file 

Noo...  it alway works for me

Jquery migrate and other wp javascripts like emoticons are removed by my plugin Speed Up - Clean WP. 

I think it's an issue how w3tc decide to move or not, javascripts to footer. 

Just for clarification, yesterday it works,  i have only update w3tc and theme

**Comment by nigrosimone** [comment]
I have fixed a similar issue in the w3tc 0.9.4.x branch https://github.com/szepeviktor/w3-total-cache-fixed/pull/248

**Comment by nigrosimone** [comment]
after investigation the problem seems `<!-- W3TC-include-js-head -->`. when this tag is in the footer w3tc don't move minified files in the footer.

**Comment by nigrosimone** [comment]
@amiga-500 if you are interested in this issue with Avada and use of `<!-- W3TC-include-js-head -->`, i have asked at the Avada's author if is possible add you in the Avada beta testing repository (normally opened only for the customers). The author is interessed in this fix because Avada recommend using w3tc to his customers. If you doesn't know Avada, Anada is the best seller theme (and maybe the most famous) for wordpress, info: https://themeforest.net/item/avada-responsive-multipurpose-theme/2833226

**Comment by nigrosimone** [comment]
> In your case, i figure  is what you need and placed in your theme's footer.php file

@amiga-500  0.9.5.x's branch missed to implement W3TC-include-js-head (0.9.4.x's feature).

This PR https://github.com/szepeviktor/w3-total-cache-fixed/pull/401 re-implement this feature in 0.9.5.x branch in the same way of 0.9.4.x branch.

However if you are still interested in the beta testing of Avada, the invitation is always available. I think make fully compatible the most famous theme with w3tc is always a good idea.

**Comment by amiga-500** [comment]
Thanks a bunch for the invite offer.  I'll consider it when things get less busy for me (i am juggling medical studies and this at the moment).  I would feel guilty taking the invite at this very moment only to have my account dormant for a while...so i will wait and let you know.  Thanks a bunch! :smile:
